cmd/api-server/cli/root.go

```diff
@@ -190,79 +232,34 @@ func runComposeServer(o *apiServerOptions, stopCh <-chan struct{}) error {
 	}
 	server.GenericAPIServer.AddPostStartHook("start-compose-server-informers", func(context genericapiserver.PostStartHookContext) error {
 		config.GenericConfig.SharedInformerFactory.Start(context.StopCh)
-		apiServiceV1beta1 := &apiregistrationv1beta1types.APIService{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "v1beta1.compose.docker.com",
-				Labels: map[string]string{
-					"com.docker.fry": "compose.api",
-				},
-			},
-			Spec: apiregistrationv1beta1types.APIServiceSpec{
-				CABundle:             caBundle,
-				Group:                v1beta1.SchemeGroupVersion.Group,
-				GroupPriorityMinimum: 1000,
-				VersionPriority:      15,
-				Version:              v1beta1.SchemeGroupVersion.Version,
-				Service: &apiregistrationv1beta1types.ServiceReference{
-					Namespace: o.serviceNamespace,
-					Name:      o.serviceName,
-				},
-			},
-		}
-
-		existing, err := aggregatorClient.APIServices().Get("v1beta1.compose.docker.com", metav1.GetOptions{})
-		if err == nil {
-			bundle, err := mergeCABundle(existing.Spec.CABundle, caBundle)
-			if err != nil {
-				return err
-			}
-			apiServiceV1beta1.ObjectMeta.ResourceVersion = existing.ObjectMeta.ResourceVersion
-			apiServiceV1beta1.Spec.CABundle = bundle
-			if _, err := aggregatorClient.APIServices().Update(apiServiceV1beta1); err != nil {
-				return err
-			}
-		} else {
-			if _, err := aggregatorClient.APIServices().Create(apiServiceV1beta1); err != nil {
-				return err
-			}
+		if err := registerAggregatedAPI(
```

Comment:
I think you can easily iterate on a `versions slice` and increment the version priority, instead of copy/paste this block 3 times 😄